### ISSUE TITLE
FinalLocalVariable doesn't report variable when condition separates 2 assignments

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#2807]
TestClass.java
````
package test;

public class TestClass {
    public void test() {
        final boolean b = true;
        int a;

        if (b) {
            a = 1;
        }
        else {
            a = 2;
        }

        System.out.println(a);
    }

    public void test2() {
        final int b = 10;
        int a;

        switch (b) {
        case 0:
            a = 1;
            break;
        default:
            a = 2;
            break;
        }

        System.out.println(a);
    }
}
````
config.xml
````
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
	<property name="charset" value="ASCII" />
	<property name="severity" value="error" />
	<property name="fileExtensions" value="java" />

	<module name="TreeWalker">
		<module name="FinalLocalVariable" />
	</module>
</module>
````
output from: `java -jar checkstyle-6.14.1-all.jar -c config.xml TestClass.java`
````
Starting audit...
Audit done.
````
Expected output:
````
Starting audit...
[ERROR] D:\Rickys\Java\EclipseStuff\checkstyle\src\main\java\test\TestClass.java:6:9: Variable 'a' should be declared final. [FinalLocalVariable]
[ERROR] D:\Rickys\Java\EclipseStuff\checkstyle\src\main\java\test\TestClass.java:20:9: Variable 'a' should be declared final. [FinalLocalVariable]
Audit done.
Checkstyle ends with 2 errors.
````

Variable `a` can be declared `final` because it is only assigned once based on the result of `b`. The compiler has no issues with it actually being `final` so checkstyle should report these instances.

**Comment by romani** [comment]
@Bhavik3 , can you help us again with this Check ?

Close relationship to #2405 isssue.

**Comment by romani** [comment]
discussion is at #2814.

fix is merged.
